### ISSUE TITLE
Raw mode with providing System.in in builder streams not Working

### ISSUE THREAD

**Issue by ju851zel** [jline/jline3#785]
When using the terminal builder and providing `System.in` and `System.out` the raw mode seems not to work.

Building the terminal like so.
```
val terminal = TerminalBuilder.terminal()
terminal.enterRawMode()
val stream = terminal.input
```
Returns every key pressed but makes it now possible to override `System.in` in Tests.

Building the terminal like so.
```
val terminal = TerminalBuilder.builder.streams(System.in, System.out).build
terminal.enterRawMode()
val stream = terminal.input
```
makes it possible to override `System.in` in tests, but does not emit every key press, but all keys when pressing enter.

Does anyone have any idea on how to use the raw mode (receive every key press) and override the input source in tests?

**Comment by gnodet** [comment]
You need to create a terminal suitable to your application or your tests, it can't be the same, especially if you need to access the system streams.  In the first case, you can override `System.in`, but the terminal will not use it because it needs access to a lower level.

For testing, you can create a custom terminal with a stream that you can use to inject data, see the following for example: https://github.com/jline/jline3/blob/a8ce4e85b850c061ecb53278b4eafdb1572e7ff5/reader/src/test/java/org/jline/reader/impl/ReaderTestSupport.java#L70-L73

Bottom line, you need to pass the `Terminal` to you application / test infrastructure instead of trying to pass an input stream.
